pkg/operator/ceph/object/spec.go

```diff
@@ -461,6 +468,20 @@ func (c *clusterConfig) makeDaemonContainer(rgwConfig *rgwConfig) (v1.Container,
 	return container, nil
 }
 
+func (c *clusterConfig) getRGWContainerCMD(args *[]string) []string {
+	if c.isLocalizedReadAffinitySet() {
+		// $hostName in the args comes from the bash script variable with the same name.
+		*args = append(*args, cephconfig.NewFlag("crush location", fmt.Sprintf("%s=%s", "host", "$hostName")))
+		parseArgsForScript(args)
+		// CrushMap HostName is parsed to replace "." with "-". `hostName=\"$(echo $NODE_NAME | tr . -)\"` does that.
+		// $NODE_NAME in the script is the node name passed to the radosgw container as downward API
+		rgwCmd := "hostName=\"$(echo $NODE_NAME | tr . -)\" && radosgw " + strings.Join(*args, " ")
+		return []string{"bash", "-x", "-c", rgwCmd}
+	}
+
+	return append([]string{"radosgw"}, *args...)
+}
```

Comment:
This suggestion combines some of my review comments into a single suggestion.

- Use bash string replacement feature instead of `tr`
- Use `exec` to ensure `radosgw` becomes the running process so we don't have issues with detecting rgw failure or passing SIGTERM/SIGKILL signals to rgw
- Use `"$@"` to pass thru args so we don't have to manage args in the script

```suggestion
func (c *clusterConfig) getRGWContainerCMD(args *[]string) []string {
	if c.isLocalizedReadAffinitySet() {
		// exec replaces the bash process with radosgw and is important to ensure signals are 
		// sent to radosgw and that the runtime detects when radosgw stops running
		// "$@" passes thru container args to radosgw process
		return []string{"bash", "-x", "-c", `
		exec radosgw --crush-location="host=${NODE_NAME//./-}" "$@"
		`
	}

	return []string{"radosgw"}
}
```

I'm 90% sure that `--crush-location="host=${NODE_NAME//./-}"` quoting is right, but it's possible this might need adjustment.